staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/schema/cel/compilation.go

```diff
@@ -139,7 +143,7 @@ func compileRule(rule apiextensions.ValidationRule, env *cel.Env) (compilationRe
 		}
 	}
 
-	prog, err := env.Program(ast, cel.EvalOptions(cel.OptOptimize))
+	prog, err := env.Program(ast, cel.EvalOptions(cel.OptOptimize, cel.OptTrackCost), cel.ActualCostLimit(perCallLimit))
```

Comment:
should the limit be min(remainingBudget, perCallLImit)?

Reply:
It is hard to get overall runtime ramainingBudget while compilation

Reply:
You're right, this is not trivial to do given how CEL is currently configured. I'm supportive of just passing in perCallLimit for now and looking into handling remainingBudget as a follow up (we should probably make Eval take a limit in cel and then use that functionality in k/k).